src/runtime.c

```diff
@@ -412,15 +413,18 @@ main (int argc, char *argv[])
                             }
                         }
                     } else if(inode.base.inode_type == SQUASHFS_REG_TYPE || inode.base.inode_type == SQUASHFS_LREG_TYPE){
-                        if(created_inode[inode.base.inode_number - 1]) {
+                        // if we've already created this inode, then this is a hardlink
+                        char* exiting_path_for_inode = created_inode[inode.base.inode_number - 1];
+                        if(exiting_path_for_inode) {
```

Comment:
Is this supposed to be a `NULL` check? Please try to use explicit comparisons, e.g., `exiting_path_for_inode == NULL`.